### PR TITLE
Fix DivideByZeroException in MonthCalendar

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- If `SingleMonthSize` is a certain size, (e.g. `new Size(6, 6)`) then a `DivideByZeroException` is thrown.
- Fix this by avoiding the `DivideByZeroException` by not checking for this case and not dividing by zero

Extracted from https://github.com/dotnet/winforms/pull/2482

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2499)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.cs [hunk 1]
```diff
@@ -1471,16 +1471,28 @@ private Size GetMinReqRect(int newDimensionLength, bool updateRows, bool updateC
 
             if (updateRows)
             {
-                Debug.Assert(calendarHeight > INSERT_HEIGHT_SIZE, "Divide by 0");
-                int nRows = (newDimensionLength - todayHeight + INSERT_HEIGHT_SIZE) / (calendarHeight + INSERT_HEIGHT_SIZE);
-                dimensions.Height = (nRows < 1) ? 1 : nRows;
+                if (calendarHeight == INSERT_HEIGHT_SIZE)
```
Comment (weltkante) [inline src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.cs]: This adds unnecessary complexity, the code is not dividing by the difference so it never matters if `calendarHeight == INSERT_HEIGHT_SIZE`

In fact you may be introducing a regression because you force `dimensions.Height = 1;` when the division might result in something else.

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.cs]: If calendarHeight <= INSERT_HEIGHT_SIZE then the result of the division will surely be either 0 or negative. This means that the check below would come in, clamping nRows to 1. By bailing out like I’ve done here and setting the dimension to 1, we don’t change the actual result

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.cs]: In terms of if there’s actually a bug, I’m not sure - we are adding INSERT_HEIGHT_SIZE to both sides of the denominator and numerator so they should outweigh each other

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.cs]: Would you prefer making my check with if (calendarHeight + INSERT_HEIGHT_SIZE) { bail } else { do the same }

Comment (weltkante) [reply src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.cs]: > If calendarHeight <= INSERT_HEIGHT_SIZE then the result of the division will surely be either 0 or negative.

Thats not true (if you didn't realize yourself yet) - the correct check would be `calendarHeight <= - INSERT_HEIGHT_SIZE`, otherwise the addition won't add up to zero or negative. This minus sign flip between the assert and actual division is suspicious because if the divisor were `calendarHeight - INSERT_HEIGHT_SIZE` instead of `calendarHeight + INSERT_HEIGHT_SIZE`  the whole thing suddenly would make sense.

Your last  change to `calendarHeight + INSERT_HEIGHT_SIZE == 0` is ok (preserves semantic of old code minus the exception) and you certainly could use `calendarHeight + INSERT_HEIGHT_SIZE <= 0` as well, though you don't have to, the `(nRows < 1) ? 1 : nRows` in the else-branch will catch that case.

TLDR: latest change looks good

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MonthCalendarTests.cs [hunk 2]
```diff
@@ -14,6 +14,11 @@ namespace System.Windows.Forms.Tests
 {
     public class MonthCalendarTests
     {
+        public MonthCalendarTests()
+        {
+            Application.ThreadException += (sender, e) => throw new Exception(e.Exception.StackTrace.ToString());
+        }
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MonthCalendarTests.cs]: Should this be expressed as a fixture?

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MonthCalendarTests.cs]: Yup - fixed
